### PR TITLE
Refactorisation in order to respect python3 project best practices

### PR DESCRIPTION
**Pull request by n3rada**
Hello to the maintainers, 👋

I appreciate your work. However, I immediately saw a few problems that would allow a python project to be scalable over time. 🐍 Thus, I spent a lot of time adjusting everything to make it installable with a simple `pipx` hit.

```shell
pipx install 'git+https://github.com/Mazars-Tech/AD_Miner.git'
```

I didn't refactor everything either, as that would take a lot of time! Maybe later, if you agree!

I took the opportunity to migrate a few repetitions to `utils` as well as improving the imports.

Of course, I tested that the report generation worked properly on the rest of my `neo4j` database remaining from a HackTheBox prolab.

I hope you'll like my participation. 😊

### CODE REVIEW

**Comment by jmbesnard** [comment]
Thanks a lot for this. I was not expecting to have contributors and certainly not less than 24 hours after the first release ! Will look into your changes next week and hopefully merge this PR ! Again, thanks a lot for the time that you have spent on this. There is definitely room for improvement in many other areas of our current code so feel free to dig further !

**Comment by n3rada** [comment]
Thank you very much! In order to test with ease and in a parallel environment, I strongly encourage you to clone my fork and run the following command:
```shell
pipx install . --suffix '-test'
```

File: README.md [hunk 1]
```diff
@@ -11,23 +11,31 @@ ADMiner was created and is maintained by the Mazars Cybersecurity Audit & Adviso
...
 
 Run the tool:
 
-    ./main.py [-h] [-b BOLT] [-u USERNAME] [-p PASSWORD] [-e EXTRACT_DATE] [-r RENEWAL_PASSWORD] [-a] [-c] [-l LEVEL] -cf CACHE_PREFIX [-ch NB_CHUNKS] [-co NB_CORES] [--rdp] [--evolution EVOLUTION] [--cluster CLUSTER]
+    AD-miner [-h] [-b BOLT] [-u USERNAME] [-p PASSWORD] [-e EXTRACT_DATE] [-r RENEWAL_PASSWORD] [-a] [-c] [-l LEVEL] -cf CACHE_PREFIX [-ch NB_CHUNKS] [-co NB_CORES] [--rdp] [--evolution EVOLUTION] [--cluster CLUSTER]
 
 Example:
 
-    ./main.py -c -cf My_Report -u neo4j -p mypassword
+    AD-minery -c -cf My_Report -u neo4j -p mypassword
```
Comment (alac88) [inline README.md:34]:
```suggestion
    AD-miner -c -cf My_Report -u neo4j -p mypassword
```

**Comment by alac88** [comment]
Thank you @n3rada for your contribution! While we are still reviewing your PR, may we ask you to rename your commits in a conventional way (`feat|fix|chore|refactor|docs`...)? This would help triggering the automatic changelog generation and a new release :)

**Comment by n3rada** [comment]
I'm clearly not going to have the time today. ⌛

My two commits are really :
1. Big global refactoring
2. Update the README.md

Can't you change what you don't like about naming yourself? I admit I don't feel like redoing my commits or factoring them into smaller commits!

The first is a big job of moving files and renaming functions, so the name "Refactoring" is common in development.

Kind regards,

**Comment by alac88** [comment]
> I'm clearly not going to have the time today. ⌛
> 
> My two commits are really :
> 
> 1. Big global refactoring
> 2. Update the README.md
> 
> Can't you change what you don't like about naming yourself? I admit I don't feel like redoing my commits or factoring them into smaller commits!
> 
> The first is a big job of moving files and renaming functions, so the name "Refactoring" is common in development.
> 
> My goal was doing this refactoring faster on purpose of introducing for you Python3 best practices at the start of the project.
> 
> Kind regards,

We do not wish to change the naming ourselves so that you will appear as a contributor. However we will still review your PR, merge it, and trigger a release on our own later on so you do not have to change anything!

Cheers

**Comment by n3rada** [comment]
Ok! Could you tell me how you would have named my commits for future ones? I'm still learning.

**Comment by alac88** [comment]
> Ok! Could you tell me how you would have named my commits for future ones? I'm still learning.

Simplest naming would be:
- refactor: huge refactoring
- docs: update README.md
See [here](https://gist.github.com/qoomon/5dfcdf8eec66a051ecd85625518cfd13) for more details

**Comment by thomas-girard** [comment]
Thank you so much for your contribution n3rada, it will greatly improve the code's quality 😀🎉 ! Just a comment about a very simple duplicate to fix:` scripts/analyse_cache.py ` appears twice in your code, once in the root directory, and the other in the "ad_miner" folder (it's a script that is not necessary for the code but can be used to read the cache if needed).

**Comment by n3rada** [comment]
My bad. I badly used `rsync` to merge the changes between the original and the fork.

**Review by thomas-girard — APPROVED** [review]

**Review by jmbesnard — APPROVED** [review]

**Review by jmbesnard — APPROVED** [review]

**Review by thomas-girard — APPROVED** [review]

**Pull request merged**